### PR TITLE
Base resource value checks enhancements

### PR DESCRIPTION
**Pull request by tronxd**
- Support of hierarchy on `BaseResourceValueCheck` checks, in order to support non-root attribute checks
- Support of multiple expected values
- Support of a default expected value, according to best practices
- Identify attributes which are variable dependant, and mark these resources as `PASSED` under the check
- Incorporation of `CKV_ANY` value,  in the case the check looks for any value for the inspected key
- Conversion of `S3Encryption`, `LaunchConfigurationEBSEncryption` and `SQSQueueEncryption` to implement the  `BaseResourceValueCheck` class

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
         id = "CKV_AWS_8"
-        supported_resources = ['aws_launch_configuration','aws_instance']
+        supported_resources = ['aws_launch_configuration', 'aws_instance']
         categories = [CheckCategories.ENCRYPTION]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
+    def get_inspected_key(self):
+        return "*_block_device/[0]/encrypted"
+
     def scan_resource_conf(self, conf):
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py:16]: @tronxd Is this method still relevant?

File: checkov/terraform/checks/resource/base_resource_value_check.py [hunk 2]
```diff
@@ -1,29 +1,92 @@
...
+
+    def get_expected_values(self):
+        """
+        Override the method with the list of acceptable values if the check has more than one possible expected value, given
+        the inspected key
+        :return: List of expected values, defaults to a list of the expected value
+        """
+        return [self.get_expected_value()]
 
     def get_expected_value(self):
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/base_resource_value_check.py:87]: @tronxd Maybe we should rename to `get_default_passing_value`?

File: checkov/common/models/consts.py [hunk 3]
```diff
@@ -1 +1,2 @@
-SUPPORTED_FILE_EXTENSIONS = [".tf", ".yml", ".yaml", ".json", ".template"]
\ No newline at end of file
+SUPPORTED_FILE_EXTENSIONS = [".tf", ".yml", ".yaml", ".json", ".template"]
+ANY_VALUE = ["CKV_ANY"]
```
Comment (nimrodkor) [inline checkov/common/models/consts.py:2]: @tronxd Would expect this to be a string

File: checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py (hunk 1 above)
Comment (schosterbarak) [reply checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py:16]: backword compatability of external checks maybe?

File: checkov/terraform/checks/resource/base_resource_value_check.py (hunk 2 above)
Comment (schosterbarak) [reply checkov/terraform/checks/resource/base_resource_value_check.py:87]: please do not rename method names - for backward compatibility with eternal checks

File: checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py (hunk 1 above)
Comment (tronxd) [reply checkov/terraform/checks/resource/aws/LaunchConfigurationEBSEncryption.py:16]: @nimrodkor @schosterbarak 
This check is a special case where the "generic" `scan_resource_conf` of `BaseResourceValueCheck` won't satisfy the check:
The check looks for 2 expected keys that share a common postfix (`root_block_device` and `ebs_block_device`, but doesn't accept a key with the same postfix  `ephemeral_block_device`
That's why the implementation of `scan_resource_conf` still remains.

File: checkov/terraform/checks/resource/base_resource_value_check.py (hunk 2 above)
Comment (tronxd) [reply checkov/terraform/checks/resource/base_resource_value_check.py:87]: @nimrodkor didn't change because of the reason above

**Pull request merged**